internal/storage/blobstorage/blob_storage.go

```diff
@@ -35,181 +17,33 @@ type (
 		Download(ctx context.Context, metadata *api.BlockMetadata) (*api.Block, error)
 	}
 
-	BlobStorageParams struct {
-		fx.In
-		fxparams.Params
-		Downloader s3.Downloader
-		Uploader   s3.Uploader
+	BlobStorageFactory interface {
```

Comment:
`blobstorage/s3/blob_storage.go`: this was my initial idea, then there was a cyclic dependency issue (blob_storage.go: factory -> implementation -> blob_storage.go: interface).

Let me take a look and find a way to improve it.

Reply:
> https://github.com/coinbase/chainsformer/tree/master/internal/controller

I see.

Reply:
Please review this again.

I'm using the name `s3` instead of `aws`, because `s3` means any s3 compatible product, e.g. minio, Cloudflare R2, DigitalOcean, etc.